### PR TITLE
[gh-452] Add basic PWA manifest

### PR DESCRIPTION
**Pull request by ashmind**

### CODE REVIEW

File: source/WebApp/build.ts [hunk 1]
```diff
@@ -81,25 +87,40 @@ const ts = task('ts', async () => {
...
-        jetpack.copyAsync(paths.from.favicon, paths.to.favicon.svg, { overwrite: true }),
+        jetpack.copyAsync(paths.from.icon, paths.to.icon.svg, { overwrite: true }),
         ...pngGeneration
     ) as unknown as Promise<void>;
-}, { inputs: [paths.from.favicon] });
+}, { inputs: [paths.from.icon] });
+
+const manifest = task('manifest', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    let content = (await jetpack.readAsync(paths.from.manifest))!;
```
Comment (jsk95) [inline source/WebApp/build.ts:109]: Hmmm tbh I think I would name the manifest manifest.template.json and leave icon out completely. Read as json not parse line by line and add icon property in to json object

Comment (ashmind) [reply source/WebApp/build.ts:109]: I did something like that at first, but it made build a bit more complicated and manifest a bit less readable as you don't see the format straight away.

I could parse as JSON to find the template I suppose — I feel attached to specific formatting, but then it should be compressed in output anyways.

Comment (jsk95) [reply source/WebApp/build.ts:109]: Pretty much. Sounds good

Comment (ashmind) [reply source/WebApp/build.ts:109]: I tried and ended up with 
```typescript
    content.icons = content.icons.flatMap(icon => {
        if (icon.src.includes('{build:each-size}'))
            return [icon];

        const parts = Object.entries(icon);
        return iconSizes.map(size => {
            const replaced = parts.map(([key, value]) => [
                key, value.replace(/\{(?:build:each-)?size\}/g, size.toString())
            ]);
            return Object.fromEntries(replaced);
        });
    });
```

Which seems more complicated (tbh JS just needs better Object.map).

Comment (ashmind) [reply source/WebApp/build.ts:109]: Hmm
```typescript
    content.icons = content.icons.flatMap(icon => {
        if (icon.src.includes('{build:each-size}'))
            return [icon];

        return iconSizes.map(size => JSON.parse(
            JSON.stringify(icon).replace(/\{(?:build:each-)?size\}/g, size.toString())
        ) as typeof icon);
    });
```

**Comment by ashmind** [comment]
🤦 `master`

**Pull request merged**